openedx_rocketchat/utils.py

```diff
@@ -0,0 +1,205 @@
+    """
+    course = re.sub('[^A-Za-z0-9]+', '', course_id)
+    room_name = "{}__{}".format("General", course)
+    response = api_rocket_chat.groups_info(room_name=room_name)
+    try:
+        response = response.json()
+
+        if response['success']:
+            api_rocket_chat.groups_invite(response['group']['_id'], user_id)
+        else:
+            kwargs = {
+                'members': [username],
+                'customFields': {
+                    'course': course,
+                    'team': None,
+                    'topic': None,
+                    'specificTeam': False,
+                }
+            }
+            api_rocket_chat.groups_create(name=room_name, **kwargs)
+
+    except AttributeError:
+        LOG.error("Create Course Group error: response with status code = %s", response.status_code)
+        pass
+
+
+def initialize_api_rocket_chat(rocket_chat_settings):
+    """
+    Returns an ApirocketChat instance using the given settings
+    """
+    admin_user = rocket_chat_settings.get('admin_user', None)
+    admin_pass = rocket_chat_settings.get('admin_pass', None)
+    url_service = rocket_chat_settings.get('public_url_service', None)
+
+    if not admin_user or not admin_pass or not url_service:
+        LOG.error('RocketChat settings error: The rocketChat credentials can not be accessed')
+        return None
+
+    try:
+        api_rocket_chat = ApiRocketChat(
+            admin_user,
+            admin_pass,
+            url_service
+        )
+    except RocketAuthenticationException:
+        LOG.error('ApiRocketChat error: RocketAuthenticationException')
+        return None
+    except RocketConnectionException:
+        LOG.error('ApiRocketChat error: RocketConnectionException')
+        return None
+
+    return api_rocket_chat
+
+
+def get_subscriptions_rids(auth_token, user_id, unread=False):
+    """
+    This method allow to get the roomid for every subscrition
+    """
+    response = user_api_rocket_chat(auth_token, user_id)._RocketChat__call_api_get('subscriptions.get')
+    try:
+        response = response.json()
+    except AttributeError:
+        return
+    if response['success']:
+        subscriptions = response.get('update', [])
+        for subscription in subscriptions:
+            if not unread:
+                yield subscription['rid']
+            elif subscription['unread'] > 0:
+                yield subscription['rid']
+
+
+def logout(auth_token, user_id):
+    """
+    Invalidate the REST API authentication token.
+    """
+    try:
+        return user_api_rocket_chat(auth_token, user_id).logout().json()
+    except AttributeError:
+        pass
+    return {"status": "fail"}
+
+
+def create_token(api_rocket_chat, user, course_key):
+    """
+    **Returns the authentication data required for rocketchat authentication process.**
+        Use Case :These credentials could be used to make api calls and rocketchat authentication on browsers.
+    """
```

Comment:
Please, make sure to specify what kind of response this function could be return, just to know what expected for calling this function. @andrey-canon

Reply:
ok